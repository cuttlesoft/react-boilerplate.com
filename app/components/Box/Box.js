/* eslint-disable react/prop-types */
/**
 *
 * Box
 *
 */

import React from 'react'
import { Box } from 'grommet'
import { doc } from './Box.doc'
import helpers from '../../utils/helpers'

function createBox({
  a11yTitle,
  align,
  alignContent,
  alignSelf,
  animation,
  as,
  background,
  basis,
  border,
  children,
  direction,
  elevation,
  fill,
  flex,
  gap,
  gridArea,
  height,
  justify,
  margin,
  overflow,
  pad,
  responsive,
  round,
  tag,
  width,
  wrap,
}) {
  return (
    <Box
      a11yTitle={a11yTitle}
      align={align}
      alignContent={alignContent}
      alignSelf={alignSelf}
      animation={animation}
      as={as}
      background={background}
      basis={basis}
      border={border}
      direction={direction}
      elevation={elevation}
      fill={fill}
      flex={flex}
      gap={gap}
      gridArea={gridArea}
      height={height}
      justify={justify}
      margin={margin}
      overflow={overflow}
      pad={pad}
      responsive={responsive}
      round={round}
      tag={tag}
      width={width}
      wrap={wrap}
    >
      {children}
    </Box>
  )
}

export default helpers.createWithDoc({
  envName: process.env.NODE_ENV,
  docFunction: doc,
  component: createBox,
})
