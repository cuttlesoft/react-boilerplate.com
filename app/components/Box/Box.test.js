import React from 'react'
import { render } from '@testing-library/react'
import 'jest-styled-components'

import Box from './Box'

const CHILDREN = <h1>Test</h1>

const renderComponent = (props = {}) => render(<Box {...props}>{CHILDREN}</Box>)

/**
 *
 * Tests for Box
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */
describe('Box', () => {
  it('renders and matches snapshot', () => {
    const { container } = renderComponent()

    expect(container).toMatchSnapshot()
  })
})
