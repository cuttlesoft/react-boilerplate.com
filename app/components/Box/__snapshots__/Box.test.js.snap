// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders and matches snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div>
  <div
    class="c0"
  >
    <h1>
      Test
    </h1>
  </div>
</div>
`;
